pkg/eventprocessor/impl/webhook.go

```diff
@@ -0,0 +1,83 @@
+package impl
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	logger "github.com/rs/zerolog/log"
+)
+
+type webhookContent struct {
+	Content string `json:"content"`
+}
+
+var webhookLogger = logger.With().Str("component", "webhook").Logger()
+
+// Common function to send the webhook request.
+func sendWebhookRequest(ctx context.Context, url string, body interface{}) error {
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	postData, err := json.Marshal(body)
+	if err != nil {
+		return fmt.Errorf("marshaling webhook JSON: %s", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewBuffer(postData))
+	if err != nil {
+		return fmt.Errorf("creating HTTP request: %s", err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	httpClient := &http.Client{
+		Timeout: 5 * time.Second,
+	}
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("executing webhook: %s", err)
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			webhookLogger.Error().Err(err).Msg("closing")
+		}
+	}()
+
+	if resp.StatusCode >= 400 {
+		return fmt.Errorf("webhook request failed with status code: %d", resp.StatusCode)
+	}
+
+	return nil
+}
+
+// Webhook interface for sending webhooks to different services such as IFTTT or Discord etc.
+type Webhook interface {
```

Comment:
Considering my comment about having just the URL we would not need different implementations. In fact, if you look at the `DiscordWebhook` implementation 

```
func (w *DiscordWebhook) Send(ctx context.Context, content string) error {
	return sendWebhookRequest(ctx, w.URL, webhookContent{
		Content: content,
	})
}
```

there's nothing in there that makes it specific to Discord, because all the Discord information is already encoded in the URL. So if you were to create another implementation, let's say Slack, it would be literally the same implementation:

```
func (w *SlackWebhook) Send(ctx context.Context, content string) error {
	return sendWebhookRequest(ctx, w.URL, webhookContent{
		Content: content,
	})
}
```

In a way, that is evidence that we don't need `EndpointType`. 

Just for completeness. It doesn't mean we don't need to treat different endpoints differently. It could be that we could need logic in the implementation to deal if the specifics of which provider. Maybe retrying logic, rate limiting handling, or error handling. And that scenario could lead to the need for different implementations. But I don't think that's the case now

Reply:
Got it! yeah, I did not realize the requests could be different and the specifics of Discord were in the `Content` fields. We would definitely need different implementations. Sounds good then to keep it. The interface you'd still need for testing.